src/prototype_room_costmatrix.js

```diff
@@ -57,41 +57,47 @@ Room.prototype.setCostMatrixPath = function(costMatrix, path) {
     let pos = path[i];
     costMatrix.set(pos.x, pos.y, config.layout.pathAvoid);
   }
+  this.setCostMatrixAvoidSources(costMatrix);
```

Comment:
yeah, but if we don't make this change, path will not avoid sources. Only first one do, but after we build some path to sourcer and overwrite costmatrix, next paths won't avoid this source